pages/_document.js

```diff
@@ -11,6 +12,7 @@ export default function Document() {
       <body>
         <Main />
         <NextScript />
+        <Script src="/theme.js" strategy="beforeInteractive" />
```

Comment:
Grabs the theme value from local storage before the DOM loads to avoid any flash of styles